File: src/main/java/mysql/hibernate/template/HibernateEntity.java
package mysql.hibernate.template;

import mysql.hibernate.EntityManagerWrapper;

public abstract class HibernateEntity implements HibernateEntityInterface {

    private EntityManagerWrapper entityManager;

    @Override
    public EntityManagerWrapper getEntityManager() {
        return entityManager;
    }

    @Override
    public void setEntityManager(EntityManagerWrapper entityManager) {
        this.entityManager = entityManager;
    }

    @Override
    public void beginTransaction() {
        entityManager.getTransaction().begin();
    }

    @Override
    public void commitTransaction() {
        entityManager.getTransaction().commit();
    }

    @Override
    public void close() {
        entityManager.close();
    }

    public void remove() {
        entityManager.remove(this);
    }

}

File: src/main/java/mysql/hibernate/template/HibernateEntity.kt
package mysql.hibernate.template

import mysql.hibernate.EntityManagerWrapper

abstract class HibernateEntity : HibernateEntityInterface {

    override lateinit var entityManager: EntityManagerWrapper

    override fun beginTransaction() {
        entityManager.transaction.begin()
    }

    override fun commitTransaction() {
        entityManager.transaction.commit()
    }

    override fun close() {
        entityManager.close()
    }

    fun remove() {
        entityManager.remove(this)
    }

    open fun postLoad() {
    }

}
